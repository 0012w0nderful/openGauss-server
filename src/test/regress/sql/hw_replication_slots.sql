--replication slots test 

select * from pg_create_physical_replication_slot('dummystandby_slot',true);
select * from pg_create_physical_replication_slot('standby_slot',false);
select * from pg_create_physical_replication_slot('standby_slot',false);

select * from pg_replication_slots order by 1;

select * from pg_drop_replication_slot('dummystandby_slot');
select * from pg_drop_replication_slot('standby_slot');
select * from pg_drop_replication_slot('wrong_slot_name');