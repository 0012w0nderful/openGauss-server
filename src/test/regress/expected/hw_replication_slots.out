--replication slots test 
select * from pg_create_physical_replication_slot('dummystandby_slot',true);
     slotname      | xlog_position 
-------------------+---------------
 dummystandby_slot | 
(1 row)

select * from pg_create_physical_replication_slot('standby_slot',false);
   slotname   | xlog_position 
--------------+---------------
 standby_slot | 
(1 row)

select * from pg_create_physical_replication_slot('standby_slot',false);
WARNING:  replication slot "standby_slot" already exists
   slotname   | xlog_position 
--------------+---------------
 standby_slot | 
(1 row)

select * from pg_replication_slots order by 1;
     slot_name     | plugin | slot_type | datoid | database | active | xmin | catalog_xmin | restart_lsn | dummy_standby 
-------------------+--------+-----------+--------+----------+--------+------+--------------+-------------+---------------
 dummystandby_slot |        | physical  |      0 |          | f      |      |              |             | t
 standby_slot      |        | physical  |      0 |          | f      |      |              |             | f
(2 rows)

select * from pg_drop_replication_slot('dummystandby_slot');
WARNING:  replicationSlotMinLSN is InvalidXLogRecPtr!!!
WARNING:  replicationSlotMaxLSN is InvalidXLogRecPtr!!!
 pg_drop_replication_slot 
--------------------------
 
(1 row)

select * from pg_drop_replication_slot('standby_slot');
WARNING:  replicationSlotMinLSN is InvalidXLogRecPtr!!!
WARNING:  replicationSlotMaxLSN is InvalidXLogRecPtr!!!
 pg_drop_replication_slot 
--------------------------
 
(1 row)

select * from pg_drop_replication_slot('wrong_slot_name');
ERROR:  replication slot "wrong_slot_name" does not exist
